geode-core/src/main/java/org/apache/geode/internal/cache/TXState.java

```diff
@@ -522,15 +522,15 @@ public void commit() throws CommitConflictException {
           // Take filter registration lock
           lockFilterRegistrationOnTxRegions();
           try {
+            attachFilterProfileInformation(entries);
+
```

Comment:
This will work with locking changes...But still feel that, the attach should be called after applying to cache...
I believe the "applyChanges" is creating the CacheEvent/EntryEvent by not setting the old value; in that case can we change that code to set the old value....If the old value is not available in apply, can we get the old value as we are doing in attach and pass it to apply?